### DISCUSSION TITLE
iOS Build Failed all of a sudden. I have no idea why.

### DISCUSSION THREAD

**Discussion by nataddrho** [codemagic-ci-cd/codemagic-docs#1645] [Q&A] [answered]
I have built about 38 versions of an app I am working on, and all of a sudden it is failing to build. I use CodeMagic because I have absolutely no understanding of how iOS apps are built, and I do not own a Mac. Did something with CodeMagic change overnight? Please help. Thanks....

My app is using flutter. What is a cocoapod?

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/11-05-23_qwyyf385.keychain-db
Searching for files matching /Users/builder/clone/digiball_app/ios/**/*.xcodeproj
Completed configuring code signing settings
 - Using profile "DigiBall Test ios_app_store 1672263227" [9b895e45-25f2-4edc-8311-efe7bba22ac5] for target "Runner" [Debug] from project "Runner"
 - Using profile "DigiBall Test ios_app_store 1672263227" [9b895e45-25f2-4edc-8311-efe7bba22ac5] for target "Runner" [Profile] from project "Runner"
 - Using profile "DigiBall Test ios_app_store 1672263227" [9b895e45-25f2-4edc-8311-efe7bba22ac5] for target "Runner" [Release] from project "Runner"
Generated options for exporting the project
 - Method: app-store
 - Provisioning Profiles:
     - net.digiball.flutterApp: DigiBall Test ios_app_store 1672263227
 - Signing Certificate: Apple Distribution
 - Signing Style: manual
 - Team Id: JVTB9JVW49
Saved export options to /Users/builder/export_options.plist

> flutter build ipa --release --export-options-plist /Users/builder/export_options.plist
Resolving dependencies...
  archive 3.3.5 (3.3.7 available)
  args 2.3.1 (2.4.1 available)
  checked_yaml 2.0.2 (2.0.3 available)
  cli_util 0.3.5 (0.4.0 available)
  crypto 3.0.2 (3.0.3 available)
  ffi 2.0.1 (2.0.2 available)
  fixnum 1.0.1 (1.1.0 available)
  flutter_blue_plus 1.3.1 (1.4.0 available)
  flutter_launcher_icons 0.11.0 (0.13.1 available)
  image 3.2.2 (4.0.17 available)
  intl 0.18.0 (0.18.1 available)
  json_annotation 4.7.0 (4.8.1 available)
  lints 2.0.1 (2.1.0 available)
  matcher 0.12.15 (0.12.16 available)
  material_color_utilities 0.2.0 (0.5.0 available)
  path_provider_linux 2.1.7 (2.1.10 available)
  path_provider_platform_interface 2.0.5 (2.0.6 available)
  path_provider_windows 2.1.3 (2.1.6 available)
  petitparser 5.1.0 (5.4.0 available)
  plugin_platform_interface 2.1.3 (2.1.4 available)
  pointycastle 3.6.2 (3.7.3 available)
  shared_preferences 2.0.15 (2.1.1 available)
  shared_preferences_android 2.0.14 (2.1.4 available)
  shared_preferences_linux 2.1.1 (2.2.0 available)
  shared_preferences_macos 2.0.4 (2.0.5 available)
  shared_preferences_platform_interface 2.1.0 (2.2.0 available)
  shared_preferences_web 2.0.4 (2.1.0 available)
  shared_preferences_windows 2.1.1 (2.2.0 available)
  source_span 1.9.1 (1.10.0 available)
  statistics 1.0.24 (1.0.25 available)
  test_api 0.5.1 (0.5.2 available)
  typed_data 1.3.1 (1.3.2 available)
  win32 3.1.2 (4.1.4 available)
  xdg_directories 0.2.0+2 (1.0.0 available)
  xml 6.1.0 (6.3.0 available)
  yaml 3.1.1 (3.1.2 available)
Got dependencies!

Archiving net.digiball.flutterApp...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: JVTB9JVW49
Running pod install...                                             428ms
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)
    [!] Unable to find a target named `RunnerTests` in project `Runner.xcodeproj`, did find `Runner`.

    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:108:in `compute_targets'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer/target_inspector.rb:40:in `compute_results'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1197:in `block (3 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1195:in `block (2 levels) in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `each'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1193:in `block in inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:1188:in `inspect_targets_to_integrate'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer/analyzer.rb:107:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:422:in `analyze'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:244:in `block in resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:243:in `resolve_dependencies'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/installer.rb:162:in `install!'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command/install.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/claide-1.1.0/lib/claide/command.rb:334:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/lib/cocoapods/command.rb:52:in `run'
    /Users/builder/.rbenv/versions/3.0.4/lib/ruby/gems/3.0.0/gems/cocoapods-1.12.1/bin/pod:55:in `<top (required)>'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `load'
    /Users/builder/.rbenv/versions/3.0.4/bin/pod:25:in `<main>'

Error output from CocoaPods:
↳

    [!] Automatically assigning platform `iOS` with version `11.0` on target `Runner` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

Error running pod install

Build failed :|
Failed to build for iOS

**Comment by nataddrho** [comment]
I did some research and I think this is a CodeMagic problem with their M1 machine.

Apparently you need to run "arch -x86_64 pod install " but I can't do this since I do not own a Mac and I am using CodeMagic's M1.

Please help!!!

**Comment by ericmand** [comment]
Check flutter version as new versions of Dart and Flutter landed during I/O that Codemagic will use if on stable.

**Comment by nataddrho** [comment]
I don't know what that means, sorry.

I reverted my GitHub repository back one revision. This revision built iOS perfectly. It is the EXACT same project, and now it doesn't build in CodeMagic.

Something changed in CodeMagic. What changed and why????

**Comment by syngetprog** [comment]
Hello.  I'm getting the same error.  Everything was fine before running this today.  Any help is appreciated.

**Reply by ericmand** [reply, marked as answer]
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/934632/54aad056-91d4-4cdb-a860-96947f0c2e18)
Codemagic defaults to Flutter Stable and Google just made Dart 3.0 included in Flutter Stable release. Dart 3.0 has lots of changes notably null safety. If things were working last week and are suddenly not working I would try running your build on an earlier Flutter release like pictured here.

**Reply by nataddrho** [reply]
Thank you. This works. I would have never figured that out.

**Comment by syngetprog** [comment]
Thanks Eric!  That worked!!

Chris

On 2023-05-11 18:12, Eric Anderson wrote:

> [1]
> Codemagic defaults to Flutter Stable and Google just made Dart 3.0 
> included in Flutter Stable release. Dart 3.0 has lots of changes 
> notably null safety. If things were working last week and are suddenly 
> not working I would try running your build on an earlier Flutter 
> release like pictured here.
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe 
> [3].
> You are receiving this because you commented.Message ID: 
> ***@***.***>


Links:
------
[1] 
https://user-images.githubusercontent.com/934632/237823855-54aad056-91d4-4cdb-a860-96947f0c2e18.png
[2] 
https://github.com/orgs/codemagic-ci-cd/discussions/1645#discussioncomment-5878390
[3]
